interp/type.go

```diff
@@ -1540,7 +1540,7 @@ func hasRecursiveStruct(t *itype, defined map[string]*itype) bool {
 		defined[typ.path+"/"+typ.name] = typ
 
 		for _, f := range typ.field {
-			if hasRecursiveStruct(f.typ, defined) {
+			if hasRecursiveStruct(f.typ, copyDefined(defined)) {
```

Comment:
I'm wondering: is it not a problem in general that copyDefined is only copying pointers of itypes instead of making deepcopies of them? is that intended?

Reply:
ah nevermind, I see we're never writing the stored itypes, it's all readonly, so it should ok.

Reply:
HasRecursive only tracks if a type refers to itself, using `defined` as an ephemeral map to mark them as visited (as in the type dependencies tree). There is no sense to deepcopy types here. But it is important that 2 sibling sub-types remain isolated, thus the `copyDefined` function.